core/config/project_settings.cpp

```diff
@@ -1200,6 +1200,7 @@ ProjectSettings::ProjectSettings() {
 	// Initialization of engine variables should be done in the setup() method,
 	// so that the values can be overridden from project.godot or project.binary.
 
+	ERR_FAIL_COND_MSG(singleton != nullptr, "Instantiating a new ProjectSettings singleton is not supported.");
```

Comment:
For the record, `memnew(ProjectSettings)` now would show this error and then crash, because failing here is not handled gracefully. I could change it to just a condition `ERR_PRINT` if preferred, though crashing here may not be bad as it's a developer mistake.

Reply:
Maybe `ERR_CRASH_COND`?

Reply:
Done.